### PR TITLE
Fix misaligned icons in 'Connected sites' modal

### LINKED ISSUE #19901: Connected sites modal misaligned icons
Within the "Connected sites" modal, the sites icons are not centered vertically.

### PR DESCRIPTION
**Pull request by jase-b**
Fixes #19901: Connected sites modal misaligned icons

Replaced `<SiteIcon>` with `<AvatarFavicon>` in `<ConnectedSitesList>`

### Before

19901 before

### After

border-muted

## Manual Testing Steps

- Log in to MetaMask.
- Connect the wallet to at least one site (if not already).
- Open the 'Connected sites' modal.
- Observe all listed site icons are centered within their circular container element.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

**Comment by jase-b** [comment]
I have read the CLA Document and I hereby sign the CLA

File: builds.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ buildTypes:
       - SEGMENT_WRITE_KEY_REF: SEGMENT_MMI_WRITE_KEY
       - SUPPORT_LINK: https://mmi-support.zendesk.com/hc/en-us
       - SUPPORT_REQUEST_LINK: https://mmi-support.zendesk.com/hc/en-us/requests/new
-      - MMI_CONFIGURATION_SERVICE_URL
+      - MMI_CONFIGURATION_SERVICE_URL: ''
```
Comment (jase-b) [inline builds.yml:92]: Default value added based on suggestion from failing CircleCI test:

mmi ppolicy test fail

File: ui/components/app/connected-sites-list/connected-sites-list.component.js [hunk 2]
```diff
@@ -33,11 +33,12 @@ export default class ConnectedSitesList extends Component {
...
-              <SiteIcon
-                icon={subject.iconUrl}
-                name={subject.name}
-                size={32}
+              <AvatarFavicon
+                borderColor="border-muted"
                 className="connected-sites-list__subject-icon"
+                name={subject.name}
+                size="sm"
+                src={subject.iconUrl}
```
Comment (NidhiKJha) [inline ui/components/app/connected-sites-list/connected-sites-list.component.js:41]:
```suggestion
              <AvatarFavicon
                borderColor={BorderColor.borderMuted}
                className="connected-sites-list__subject-icon"
                name={subject.name}
                size={Size.MD}
                src={subject.iconUrl}
```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
The changes look really great. I have suggested some changes to use border color and size from design-system constants. Thanks for your contribution @jase-b

**Review by NidhiKJha — APPROVED** [review]
LGTM! Thanks for your contribution

**Review by vthomas13 — APPROVED** [review]

**Comment by vthomas13** [comment]
Congrats @jase-b! Really appreciate the contribution 😁

**Pull request merged**